### ISSUE TITLE
build : rook image build failure due to vault sdk invalid version

### ISSUE THREAD

**Issue by thotz** [rook/rook#6583]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Rook image building fails due to an invalid version of vault-sdk

```console

# make mod.update
=== updating modules
go: github.com/libopenstorage/secrets@v0.0.0-20201006135900-af310b01fe47 requires
	github.com/hashicorp/vault@v1.4.2 requires
	github.com/hashicorp/go-kms-wrapping@v0.5.1 requires
	github.com/hashicorp/vault/sdk@v0.1.14-0.20191229212425-c478d00be0d6: invalid version: unknown revision c478d00be0d6
make: *** [build/makelib/golang.mk:166: go.mod.update] Error 1
```
libopenstorage points to vault library 1.4.2, which uses invalid version of sdk, at this issue https://github.com/hashicorp/go-kms-wrapping/issues/19 mentions similar issue exists with 1.4.3

**Expected behavior:**
Image build should work
**How to reproduce it (minimal and precise):**

* clone new rook repo
* build rook image using make command

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by thotz** [comment]
@leseb : should we update the vault version in libopenstorage to v1.5??. Will that okay for rook ?

**Comment by leseb** [comment]
@thotz I don't understand, Rook builds correctly, why are you running `make mod.update`?

**Comment by subhamkrai** [comment]
> @thotz I don't understand, Rook builds correctly, why are you running `make mod.update`?

@leseb PTAL on this discussion https://rook-io.slack.com/archives/C46Q5UC05/p1604520842311000.

**Comment by leseb** [comment]
> > @thotz I don't understand, Rook builds correctly, why are you running `make mod.update`?
> 
> @leseb PTAL on this discussion https://rook-io.slack.com/archives/C46Q5UC05/p1604520842311000.

Ok but it looks like you gave the solution, didn't you?

**Comment by subhamkrai** [comment]
> > > @thotz I don't understand, Rook builds correctly, why are you running `make mod.update`?
> > 
> > 
> > @leseb PTAL on this discussion https://rook-io.slack.com/archives/C46Q5UC05/p1604520842311000.
> 
> Ok but it looks like you gave the solution, didn't you?

Yes, and I also faced but that solution worked.

**Comment by leseb** [comment]
Here's the solution:

> You need to update your `
> go env GOPROXY
> https://proxy.golang.org,direct
> `by adding in .bashrc (in lilnux). See this https://github.com/hashicorp/go-kms-wrapping/issues/19

**Comment by thotz** [comment]
@leseb @subhamkrai: I agree that changing env GOPROXY will resolve this issue, but it likes workaround or hacky fix even [gokmswrapping/issue#19](https://github.com/hashicorp/go-kms-wrapping/issues/19) says the same IMO. Actually fix is point `vault/sdk` into right tag. I made that change in https://github.com/rook/rook/pull/6602. But it resulted updating other modules in the rook repo

**Comment by subhamkrai** [comment]
@thotz Not sure which is the right way of doing but yes, reading comments from that particular issues seems like it's hacky way.
